java/code/src/com/redhat/rhn/taskomatic/task/RhnJavaJob.java

```diff
@@ -31,37 +33,45 @@
 import java.util.Arrays;
 
 
-
 /**
  * RhnJavaJob
  */
 public abstract class RhnJavaJob implements RhnJob {
 
+    protected static String DEFAULT_PLUGIN_NAME = "File";
     protected Logger log = LogManager.getLogger(getClass());
 
     protected Logger getLogger() {
         return log;
     }
 
-    void enableLogging(TaskoRun run) {
+    protected void enableLogging(TaskoRun run) {
+
+        var loggerName = this.getClass().getName();
+        var ctx = (LoggerContext) LogManager.getContext(false);
+        final Configuration config = ctx.getConfiguration();
+        for (var e : config.getLoggerConfig(loggerName).getAppenders().entrySet()) {
+            config.getLoggerConfig(loggerName).removeAppender(e.getKey());
+        }
+        Configurator.reconfigure(config);
```

Comment:
Why are you reconfiguring here and then at the very last line of this method again?

Reply:
The config here and the one down below seem to be different objects. I think it makes sense to call it twice for the changes to take effect. But I'm not 100% sure